manta-pay/src/test/compat.rs

```diff
@@ -0,0 +1,307 @@
+        &spending_key_1,
+        asset_1,
+        rng,
+    )
+    .expect("Unable to build MINT proof.");
+    assert_valid_proof(&verifying_context.mint, &mint_1);
+    let sender_1 = pre_sender_1
+        .insert_and_upgrade(&mut utxo_accumulator)
+        .expect("Just inserted so this should not fail.");
+    let private_transfer = PrivateTransfer::build(
+        [sender_0, sender_1],
+        [
+            spending_key_0.receiver(parameters, rng.gen(), asset_1),
+            spending_key_1.receiver(parameters, rng.gen(), asset_0),
+        ],
+    )
+    .into_post(
+        FullParameters::new(parameters, utxo_accumulator.model()),
+        &proving_context.private_transfer,
+        rng,
+    )
+    .expect("Unable to build PRIVATE_TRANSFER proof.");
+    assert_valid_proof(&verifying_context.private_transfer, &private_transfer);
+    ([mint_0, mint_1], private_transfer)
+}
+
+/// Samples a [`Reclaim`] transaction under two [`Mint`]s.
+#[inline]
+fn sample_reclaim<R>(
+    proving_context: &MultiProvingContext,
+    verifying_context: &MultiVerifyingContext,
+    parameters: &Parameters,
+    utxo_accumulator_model: &UtxoAccumulatorModel,
+    asset_0: Asset,
+    asset_1: Asset,
+    rng: &mut R,
+) -> ([TransferPost; 2], TransferPost)
+where
+    R: CryptoRng + RngCore + ?Sized,
+{
+    let mut utxo_accumulator = UtxoAccumulator::new(utxo_accumulator_model.clone());
+    let spending_key_0 = SpendingKey::new(rng.gen(), rng.gen());
+    let (mint_0, pre_sender_0) = transfer::test::sample_mint(
+        &proving_context.mint,
+        FullParameters::new(parameters, utxo_accumulator.model()),
+        &spending_key_0,
+        asset_0,
+        rng,
+    )
+    .expect("Unable to build MINT proof.");
+    assert_valid_proof(&verifying_context.mint, &mint_0);
+    let sender_0 = pre_sender_0
+        .insert_and_upgrade(&mut utxo_accumulator)
+        .expect("Just inserted so this should not fail.");
+    let spending_key_1 = SpendingKey::new(rng.gen(), rng.gen());
+    let (mint_1, pre_sender_1) = transfer::test::sample_mint(
+        &proving_context.mint,
+        FullParameters::new(parameters, utxo_accumulator.model()),
+        &spending_key_1,
+        asset_1,
+        rng,
+    )
+    .expect("Unable to build MINT proof.");
+    assert_valid_proof(&verifying_context.mint, &mint_1);
+    let sender_1 = pre_sender_1
+        .insert_and_upgrade(&mut utxo_accumulator)
+        .expect("Just inserted so this should not fail.");
+    let reclaim = Reclaim::build(
+        [sender_0, sender_1],
+        [spending_key_0.receiver(parameters, rng.gen(), asset_1)],
+        asset_0,
+    )
+    .into_post(
+        FullParameters::new(parameters, utxo_accumulator.model()),
+        &proving_context.reclaim,
+        rng,
+    )
+    .expect("Unable to build RECLAIM proof.");
+    assert_valid_proof(&verifying_context.reclaim, &reclaim);
+    ([mint_0, mint_1], reclaim)
+}
```

Comment:
These functions are also used in [manta-benchmark](https://github.com/Manta-Network/manta-rs/blob/main/manta-benchmark/src/payment.rs#L53-L164). Maybe we only need one copy.